Mathlib/Data/Matrix/Rank.lean

```diff
@@ -18,17 +18,87 @@ This definition does not depend on the choice of basis, see `Matrix.rank_eq_finr
 ## Main declarations
 
 * `Matrix.rank`: the rank of a matrix
+* `Matrix.cRank`: the rank of a matrix as a cardinal
+* `Matrix.eRank`: the rank of a matrix as a term in `ℕ∞`.
 
 -/
 
-
 open Matrix
 
 namespace Matrix
 
-open Module
+open Module Cardinal Set Submodule
+
+universe ul um um₀ un un₀ uo uR
+variable {l : Type ul} {m : Type um} {m₀ : Type um₀} {n : Type un} {n₀ : Type un₀} {o : Type uo}
+variable {R : Type uR}
+
+section Infinite
+
+variable [Semiring R]
+
+/-- The rank of a matrix, defined as the dimension of its column space, as a cardinal. -/
+noncomputable def cRank (A : Matrix m n R) : Cardinal := Module.rank R <| span R <| range Aᵀ
+
+lemma cRank_toNat_eq_finrank (A : Matrix m n R) :
+    A.cRank.toNat = Module.finrank R (span R (range Aᵀ)) := rfl
+
+lemma lift_cRank_submatrix_le (A : Matrix m n R) (r : m₀ → m) (c : n₀ → n) :
+    lift.{um} (A.submatrix r c).cRank ≤ lift.{um₀} A.cRank := by
+  have h : ((A.submatrix r id).submatrix id c).cRank ≤ (A.submatrix r id).cRank :=
+    Submodule.rank_mono <| span_mono <| by rintro _ ⟨x, rfl⟩; exact ⟨c x, rfl⟩
+  refine (Cardinal.lift_monotone h).trans ?_
+  let f : (m → R) →ₗ[R] (m₀ → R) := LinearMap.funLeft R R r
+  have h_eq : Submodule.map f (span R (range Aᵀ)) = span R (range (A.submatrix r id)ᵀ) := by
+    rw [LinearMap.map_span, ← image_univ, image_image, transpose_submatrix]
+    aesop
+  rw [cRank, ← h_eq]
+  have hwin := lift_rank_map_le f (span R (range Aᵀ))
+  simp_rw [← lift_umax] at hwin ⊢
+  exact hwin
+
+/-- A special case of `lift_cRank_submatrix_le` for when `m₀` and `m` are in the same universe. -/
+lemma cRank_submatrix_le {m m₀ : Type um} (A : Matrix m n R) (r : m₀ → m) (c : n₀ → n) :
+    (A.submatrix r c).cRank ≤ A.cRank := by
+  simpa using lift_cRank_submatrix_le A r c
+
+lemma cRank_le_card_row [StrongRankCondition R] [Fintype m] (A : Matrix m n R) :
```

Comment:
This doesn't match `rank_le_card_width` further down the file.

I think `card_row` is misleading, since the cardinality of a row could be interepreted as the number of columns!

If you really do feel this new name is better, then please:
* Explain why in the PR description
* Rename and deprecate the `rank` lemmas using the old convention

Reply:
Good point. Will change it after kid bedtime.